rust-mode.el

```diff
@@ -1195,12 +1195,24 @@ This is written mainly to be used as `end-of-defun-function' for Rust."
   (with-current-buffer (get-buffer-create "*rustfmt*")
     (erase-buffer)
     (insert-buffer-substring buf)
-    (if (zerop (call-process-region (point-min) (point-max) rust-rustfmt-bin t t nil))
-        (progn
-          (if (not (string= (buffer-string) (with-current-buffer buf (buffer-string))))
+    (let ((tmpf (make-temp-file "rustfmt")))
```

Comment:
It seems like it would be better to wrap the the body here in an `unwind-protect` that removes the temporary file.